group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2227,81 +2611,95 @@ public CoordinatorResult<Void, Record> classicGroupJoin(
         RequestContext context,
         JoinGroupRequestData request,
         CompletableFuture<JoinGroupResponseData> responseFuture
+    ) {
+        throwIfClassicGroupSessionTimeoutInvalid(request.sessionTimeoutMs());
+
+        Group group = groups.get(request.groupId(), Long.MAX_VALUE);
+        if (group != null && group.type() == CONSUMER && !group.isEmpty()) {
```

Comment:
The `!group.isEmpty()` condition is a bit subtile to grasp at first. My understanding is that we want empty consumer groups to be converted to classic groups. Is my understanding correct? If it is, it may be good to add a small comment to the `if` branch to explain it.